src/main/java/ca/uhn/fhir/jpa/starter/FhirServerConfigR4.java

```diff
@@ -81,4 +81,8 @@ public ElasticsearchSvcImpl elasticsearchSvc() {
     }
   }
 
+  @Bean
+  public CollectDataResourceProvider collectDataResourceProvider() {
+      return new CollectDataResourceProvider();
+  }
```

Comment:
Instead of this, I think it's easier to just annotate the `CollectDataResourceProvider` as a `@Component` no? The main reason for using the `@Bean` annotation is when we need to do some configuration other than just "here's the class for this".

Reply:
sure , done

Reply:
@ibacher  , to fix tests for other FHIR contexts ,
i had to remove the @component annotation ,
and instead place the @Bean configuration in the https://github.com/openmrs/openmrs-contrib-plm-fhir-server/blob/master/src/main/java/ca/uhn/fhir/jpa/starter/FhirServerConfigCommon.java  class , so that Spring can load it for even  other FHIR CONTEXTS other than R4.
Other wise other Test tests for other FHIR contexts were failing ,and only R4 Tests passing